src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int EMPTY_SIZE = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value, last, null);
+        if (last == null) {
+            first = node;
+        } else {
+            last.next = node;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAddByIndex(index);
+        if (index >= size) {
```

Comment:
Instead of 'if (index >= size)', it's better to use 'if (index == size)' to make it clear that we are handling the case where the index is equal to the size (i.e., adding to the end of the list).